src/Driver/PgSQL/Result.php

```diff
@@ -93,7 +95,7 @@ public function fetchAllNumeric(): array
 
         return array_map(
             fn (array $row) => $this->mapNumericRow($row, $types),
-            pg_fetch_all($this->result, PGSQL_NUM),
+            $this->pgFetchAll($this->result, PGSQL_NUM),
```

Comment:
I don't think that introducing this method helps us in any way.

Reply:
It's just a way to type the second parameter for `array_map`, extracted to a function because it was being called a couple of times. The alternative would be

```php
/** @var array<non-empty-list<string|null>> $fetchAll */
$fetchAll = pg_fetch_all($this->result, PGSQL_NUM);

return array_map(
    fn (array $row) => $this->mapNumericRow($row, $types),
    $fetchAll,
);
```

Reply:
> it was being called a couple of times

"a couple of times" would be exactly twice here. And you've even added a conditional return type that handles each of those two calls differently. That's unnecessarily complicated, imho.

Let's go with your alternative.